atom/browser/auto_updater_mac.mm

```diff
@@ -104,8 +104,8 @@
           delegate->OnUpdateNotAvailable();
         }
       } error:^(NSError *error) {
-        NSMutableString* failureString =
-          [NSMutableString stringWithString:error.localizedDescription];
+        NSMutableString *failureString =
+          [[NSString stringWithFormat:@"%@:%@:%@", error.code, error.domain, error.localizedDescription] mutableCopy];
```

Comment:
Thinking whether or not this is a breaking change if people were previously using this to determine the error.  Might be a safer solution to set `code` and `domain` as attributes on the emitted error object.

You can do that pretty easily in `atom_api_auto_updater.cc`.  Just need to change the `OnError` method to accept a code and a message as well.

```cpp
void AutoUpdater::OnError(const std::string& message, const int code, const std::string& domain) {
  v8::Locker locker(isolate());
  v8::HandleScope handle_scope(isolate());
  auto error = v8::Exception::Error(mate::StringToV8(isolate(), message));
  auto errorObject = error->ToObject(isolate()->GetCurrentContext()).ToLocalChecked();
  errorObject->Set(mate::StringToV8(isolate(), "code"), v8::Integer::New(isolate(), code));
  errorObject->Set(mate::StringToV8(isolate(), "domain"), mate::StringToV8(isolate(), domain));
  mate::EmitEvent(
      isolate(),
      GetWrapper(),
      "error",
      errorObject,
      // Message is also emitted to keep compatibility with old code.
      message);
}
```